apps/omg/lib/omg/output/uniqueness_enforcer.ex

```diff
@@ -0,0 +1,47 @@
+# Copyright 2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.Output.UniquenessEnforcer do
+  @moduledoc """
+  Output to enforce transaction's hash uniqueness based on plasma block number.
```

Comment:
elaborate more pls. Actually haven't re-read the design, I have no idea what this output is for :joy:

Reply:
I think this is to solve the problem of multiple fee transactions resulting in the same hashes because they're basically the same transactions in different blocks. 

Quoting Andy here:

> The main reason is to find a way to make sure the tx would be unique. Without that, any two same token fee tx with the same amount would result in the same tx bytes. We had a lot of trouble and treated it as a special case when deposit tx had this problem

Reply:
We are changing the mechanism as proposed in this PR: https://github.com/omisego/feat-fee/pull/47

So we will put a nonce into the transaction instead of appending output.